src/java/com/zimbra/cs/taglib/tag/LoginTag.java

```diff
@@ -163,9 +165,14 @@ public void doTag() throws JspException, IOException {
                     String protocol = (ZJspSession.isProtocolModeHttps() ? "httpssl" : "http");
                     NginxAuthServer nginxLookUpServer = NginxRouteLookUpConnector.getClient().getRouteforAccount(mUsername, "username",
                             protocol, HttpUtil.getVirtualHost(request), request.getRemoteAddr(), request.getHeader("Virtual-Host"));
-                    // In case of https, protocol needs to be https for the URL and not httpssl as passed to getRouteforAccount
-                    protocol = (ZJspSession.isProtocolModeHttps() ? "https" : "http");
-                    mUrl = protocol + "://" + nginxLookUpServer.getNginxAuthServer() + "/service/soap";
+                    if (null == nginxLookUpServer) {
+                        throw new SoapFaultException("Nginx route lookup error: Authentication failed for [" + mUsername + "]",
+                                AccountServiceException.AUTH_FAILED, false, null);
+                    } else {
+                        // In case of https, protocol needs to be https for the URL and not httpssl as passed to getRouteforAccount
+                        protocol = (ZJspSession.isProtocolModeHttps() ? "https" : "http");
+                        mUrl = protocol + "://" + nginxLookUpServer.getNginxAuthServer() + "/service/soap";
+                    }
                 } else {
                     mUrl = ZJspSession.getSoapURL(pageContext);
```

Comment:
`getSoapURL` also fetches an nginx auth server and uses without checking for null. This is used in a few other tags too - do you need to consider that implementation?

Reply:
No, I don't think we need to take care of the case in the getSoapURL() method.  

In the getSoapURL() method, the NginxAuthServer object is fetched only if the account is registered in the LDAP server.  And if the account is registered in the LDAP, the NginxAuthServer object will be always not-NULL value.  (The only case which NginxAuthServer object becomes NULL is when the LDAP doesn't return the 200 status code or doesn't return "Auth-Status: OK" in the response header; which are some serious problem and they should be throw the exception).  So we don't have to worry about the NULL for the NginxAuthServer in the getSoapURL().